src/esw/brushless.py

```diff
@@ -0,0 +1,231 @@
+        self.torque = max(0, min(CommandData.MAX_TORQUE, torque))
+
+
+class MoteusBridge:
+    def __init__(self, can_id: int):
+        self.CAN_ID = can_id
+        self.last_updated_time = t.time()
+        self.controller = moteus.Controller(id=can_id)
+        self.state = "Disconnected"
+        self.command_lock = threading.Lock()
+        self._command = CommandData(position=math.nan, velocity=0.0, torque=CommandData.MAX_TORQUE)
+        self._fault_response = 0
+        self.TIME_INDICATING_DISCONNECTED = 0.01
+        self.WATCHDOG_LIMIT = 0.1
+
+    def set_command(self, command: CommandData) -> None:
+        self.command_lock.acquire()
+        self._command = command
+        self.command_lock.release()
+
+    async def _send_command(self) -> None:
+        self.command_lock.acquire()
+        command = self._command
+        self.command_lock.release()
+        try:
+            state = await asyncio.wait_for(
+                self.controller.set_position(
+                    position=command.position,
+                    velocity=command.velocity,
+                    velocity_limit=CommandData.VELOCITY_LIMIT,
+                    maximum_torque=command.torque,
+                    watchdog_timeout=self.WATCHDOG_LIMIT,
+                    query=True,
+                ),
+                timeout=self.TIME_INDICATING_DISCONNECTED,
+            )
+            self._fault_response = state.values[moteus.Register.FAULT]
+        except asyncio.TimeoutError:
+            if self.state != "Disconnected":
+                rospy.logerr(f"CAN ID {self.CAN_ID} disconnected when trying to send command")
+            self.state = "Disconnected"
+
+    def _has_error(self) -> bool:
+        return self._fault_response != 0
+
+    async def _connect(self) -> None:
+        try:
+            await asyncio.wait_for(self.controller.set_stop(), timeout=0.1)
+            await asyncio.wait_for(
+                self.controller.set_position(
+                    position=math.nan,
+                    velocity=0,
+                    velocity_limit=CommandData.VELOCITY_LIMIT,
+                    maximum_torque=CommandData.MAX_TORQUE,
+                    watchdog_timeout=self.WATCHDOG_LIMIT,
+                    query=True,
+                ),
+                timeout=self.TIME_INDICATING_DISCONNECTED,
+            )
+        except asyncio.TimeoutError:
+            if self.state != "Disconnected":
+                rospy.logerr(f"CAN ID {self.CAN_ID} disconnected when trying to connect")
+            self.state = "Disconnected"
+            return
+        self.state = "Armed"
+
+    async def update(self) -> None:
+        if self.state == "Armed":
+            errors = self._has_error()
+
+            if errors:
+                rospy.logerr(f"CAN ID {self.CAN_ID} has error {self._fault_response}")
+                self.state = "Error"
+                return
```

Comment:
I made an error codes dictionary on my branch for this based on the one in the moteus library. Why is it not being used here? You can very easily translate the error number into the meaning. (ie. fault response of 33 means it's a motor driver fault)